x-pack/plugins/fleet/server/services/app_context.ts

```diff
@@ -172,6 +174,23 @@ class AppContextService {
     }
     return this.savedObjectsTagging;
   }
+  public getInternalUserSOClientForSpaceId(spaceId?: string) {
+    const request = CoreKibanaRequest.from({
+      headers: {},
+      path: '/',
+      route: { settings: {} },
+      url: { href: '', hash: '' } as URL,
+      raw: { req: { url: '/' } } as any,
+    });
+    if (this.httpSetup && spaceId && spaceId !== DEFAULT_SPACE_ID) {
+      this.httpSetup?.basePath.set(request, `/s/${spaceId}`);
```

Comment:
should the `basePath` be set also when `spaceId` is default?

Reply:
it seems it's not needed when the space is default, we really only use this as a way to set the space on the request.